de-lib/src/main/java/org/iplantc/de/admin/apps/client/presenter/grid/AdminAppsGridPresenterImpl.java

```diff
@@ -427,4 +481,34 @@ Splittable convertAppToSplittable(App app) {
     App convertSplittableToApp(Splittable appSpl) {
         return AutoBeanCodex.decode(factory, App.class, appSpl.getPayload()).as();
     }
+
+    List<App> SplittableToAppList(Splittable apps) {
+        return AutoBeanCodex.decode(factory, AppList.class, apps).as().getApps();
+    }
+
+    List<App> removeAppFromListing(String id) {
+        if (apps != null) {
+            List<App> appList = SplittableToAppList(apps);
+            if (appList != null) {
+                List<App> newAppList = appList.stream()
+                                              .filter(app -> !app.getId().equals(id))
+                                              .collect(Collectors.toList());
+                return newAppList;
+            }
+            return null;
+        }
+        return null;
+    }
+
+    void updateApp(App updatedApp) {
+        if (apps != null) {
+            List<App> newAppList = removeAppFromListing(updatedApp.getId());
+            if (newAppList != null) {
+                AppList appListBean = factory.appList().as();
+                newAppList.add(updatedApp);
+                appListBean.setApps(newAppList);
+                setApps(AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(appListBean)));
+            }
+        }
+    }
```

Comment:
I know I advocated strongly against doing splittable to Autobean conversion (and vice-versa) but I am afraid there is not a straightforward way to figure out where the app listing is updated from and refresh that list.